rust_src/src/buffers.rs

```diff
@@ -575,18 +577,18 @@ impl LispBufferOrName {
 impl From<LispBufferOrName> for LispObject {
     fn from(buffer_or_name: LispBufferOrName) -> Self {
         match buffer_or_name {
-            LispBufferOrName::Buffer(b) => b,
-            LispBufferOrName::Name(n) => n,
+            LispBufferOrName::Buffer(b) => b.into(),
+            LispBufferOrName::Name(n) => n.into(),
         }
     }
 }
 
 impl From<LispObject> for LispBufferOrName {
     fn from(v: LispObject) -> Self {
         if v.is_string() {
```

Comment:
Since you want the LispString type, why not use `if let Some(s) = v.as_string()` style? No need to check the string type twice which will happen with the `is_string` and then `Into<LispString>` calls.

Same with the LispBuffer below.